geode-core/src/integrationTest/java/org/apache/geode/cache/client/internal/QueueManagerJUnitTest.java

```diff
@@ -294,6 +298,84 @@ public void testAddToConnectionListCallsCloseConnectionOpWithKeepAliveTrue2() {
+
+    createRegisterInterestTracker(pool, testRegion);
+
+    manager = new QueueManagerImpl(pool, endpoints, source, factory, 2,
+        20, logger, ClientProxyMembershipID.getNewProxyMembership(ds));
+    manager.start(background);
+    manager.setSendClientReady();
+    manager.clearQueueConnections();
+    factory.addConnection(0, 0, 4);
+    manager.recoverPrimary(excludedServers);
+
+    assertThat(opList.get(0)).isInstanceOf(RegisterInterestListOp.RegisterInterestListOpImpl.class);
+    assertThat(opList.get(1)).isInstanceOf(RegisterInterestListOp.RegisterInterestListOpImpl.class);
+    assertThat(opList.get(2)).isInstanceOf(RegisterInterestListOp.RegisterInterestListOpImpl.class);
+    assertThat(opList.get(3)).isInstanceOf(RegisterInterestListOp.RegisterInterestListOpImpl.class);
+    assertThat(opList.get(4)).isInstanceOf(ReadyForEventsOp.ReadyForEventsOpImpl.class);
+  }
+
+  private void createRegisterInterestTracker(InternalPool localPool,
+      LocalRegion localRegion) {
+    final RegisterInterestTracker registerInterestTracker = localPool.getRITracker();
+
+    final ConcurrentHashMap<String, RegisterInterestTracker.RegionInterestEntry> keysConcurrentMap =
+        new ConcurrentHashMap<>();
+    when(registerInterestTracker.getRegionToInterestsMap(eq(InterestType.KEY), anyBoolean(),
+        anyBoolean())).thenReturn(
+            keysConcurrentMap);
+    RegisterInterestTracker.RegionInterestEntry registerInterestEntry =
+        new RegisterInterestTracker.RegionInterestEntry(
+            localRegion);
+
+    registerInterestEntry.getInterests().put("bob", InterestResultPolicy.NONE);
+    keysConcurrentMap.put("testRegion", registerInterestEntry);
+
+    final ConcurrentHashMap<String, RegisterInterestTracker.RegionInterestEntry> regexConcurrentMap =
+        new ConcurrentHashMap<>();
+    when(registerInterestTracker.getRegionToInterestsMap(eq(InterestType.REGULAR_EXPRESSION),
+        anyBoolean(), anyBoolean())).thenReturn(
+            regexConcurrentMap);
+
+    final ConcurrentHashMap<String, RegisterInterestTracker.RegionInterestEntry> filterClassConcurrentMap =
+        new ConcurrentHashMap<>();
+    when(registerInterestTracker.getRegionToInterestsMap(eq(InterestType.FILTER_CLASS),
+        anyBoolean(), anyBoolean())).thenReturn(
+            filterClassConcurrentMap);
+
+    final ConcurrentHashMap<String, RegisterInterestTracker.RegionInterestEntry> cqConcurrentMap =
+        new ConcurrentHashMap<>();
+    when(registerInterestTracker.getRegionToInterestsMap(eq(InterestType.CQ), anyBoolean(),
+        anyBoolean())).thenReturn(
+            cqConcurrentMap);
+
+    final ConcurrentHashMap<String, RegisterInterestTracker.RegionInterestEntry> oqlQueryConcurrentMap =
+        new ConcurrentHashMap<>();
+    when(registerInterestTracker.getRegionToInterestsMap(eq(InterestType.OQL_QUERY), anyBoolean(),
+        anyBoolean())).thenReturn(
+            oqlQueryConcurrentMap);
```

Comment:
This can be simplified considerably to:
```
    for (InterestType interestType : InterestType.values()) {
      final ConcurrentHashMap<String, RegisterInterestTracker.RegionInterestEntry> concurrentMap =
          new ConcurrentHashMap<>();
      when(registerInterestTracker.getRegionToInterestsMap(eq(interestType), anyBoolean(),
          anyBoolean()))
          .thenReturn(concurrentMap);
      if (interestType.equals(InterestType.KEY))  {
        RegisterInterestTracker.RegionInterestEntry registerInterestEntry =
            new RegisterInterestTracker.RegionInterestEntry(localRegion);

        registerInterestEntry.getInterests().put("bob", InterestResultPolicy.NONE);
        concurrentMap.put("testRegion", registerInterestEntry);
      }
    }
```

Reply:
Done. Thanks!